Specifications/QIR/Data-Types.md

```diff
@@ -264,6 +287,8 @@ an array should cause an immediate runtime failure.
 This applies to slicing and projection operations as well as to element access.
 When validating indices for slicing, only indices that are actually part of the
 resolved range should be considered.
+When projecting, an attempt to project a one-dimensional array on its only dimension
```

Comment:
Unless I am misunderstanding what you mean by projecting on its only dimension: why shouldn't the projection on its only dimension return the array itself?

Reply:
Projection must decrease array's dimensionality and we don't support 0-dimentional arrays. Maybe we could treat scalars as such but is it worth the complexity?

Reply:
I think I am just tripping up on the formulation; thinking geometrically, to remove one dimension, you project onto the perpendicular hyperspace. Saying that it is projected onto its dimension sounds like it is projected on the axis it is already on in the 1D case.

Reply:
Maybe I should clean up the wording; geometrically you're always projecting onto the perpendicular hyperspace of the selected dimension. Maybe projecting "along" rather than "on"?

Reply:
Yes, I think that makes sense.